src/audio/code_8019AF00.c

```diff
@@ -4211,46 +4265,246 @@ void AudioSfx_LowerSfxSettingsReverb(Vec3f* pos, s8 isReverbLowered) {
+ *     - volume will decrease by 0.0375f
+ *     - frequency will decrease by 0.0333333f
+ */
+void Audio_PlaySfx_AtPosForMetalEffectsWithSyncedFreqAndVolume(Vec3f* pos, u16 sfxId, f32 freqVolParam) {
+    f32 sp2C;
+    f32 phi_f0;
+    s32 phi_v0;
+    u16 metalSfxId = 0;
+
+    sp2C = Audio_SetSyncedSfxFreqAndVolume(freqVolParam);
+    AudioSfx_PlaySfx(sfxId, pos, 4, &sSfxSyncedFreq, &sSfxSyncedVolume, &gSfxDefaultReverb);
+
+    if ((sfxId & 0xF0) == 0xB0) {
+        // Crawlspaces (unused remnant of OoT)
+        phi_f0 = 0.3f;
+        phi_v0 = true;
+        sp2C = 1.0f;
+    } else {
+        phi_f0 = 1.1f;
+        phi_v0 = gAudioCtx.audioRandom & 1 & 0xFF;
+    }
+
+    if (phi_f0 < freqVolParam) {
+        if (phi_v0) {
+            if ((sfxId & 0x1FF) < 0x80) {
+                metalSfxId = NA_SE_PL_METALEFFECT_KID;
+            } else if ((sfxId & 0x1FF) < 0xF0) {
+                metalSfxId = NA_SE_PL_METALEFFECT_ADULT;
+            }
+
+            if (metalSfxId != 0) {
+                sSfxSyncedVolumeForMetalEffects = (sp2C * 0.7) + 0.3;
+                AudioSfx_PlaySfx(metalSfxId, pos, 4, &sSfxSyncedFreq, &sSfxSyncedVolumeForMetalEffects,
+                                 &gSfxDefaultReverb);
+            }
+        }
+    }
+}
+
+/**
+ * Adjusts both frequency and volume based on a single parameter "freqVolParam"
+ * When freqVolParam >= 6.0f, frequency is increased to 1.1f and volume remains fixed at 1.0f
+ * For every -1.0f taken from freqVolParam (eg. 5.0f, 4.0f, 3.0f...):
+ *     - volume will decrease by 0.0375f
+ *     - frequency will decrease by 0.0333333f
+ */
+void Audio_PlaySfx_AtPosWithSyncedFreqAndVolume(Vec3f* pos, u16 sfxId, f32 freqVolParam) {
+    Audio_SetSyncedSfxFreqAndVolume(freqVolParam);
+    AudioSfx_PlaySfx(sfxId, pos, 4, &sSfxSyncedFreq, &sSfxSyncedVolume, &gSfxDefaultReverb);
+}
+
+void Audio_PlaySfx_GiantsMaskUnused(Vec3f* pos, u16 sfxId) {
+    AudioSfx_PlaySfx(sfxId | 0xE0, pos, 4, &sGiantsMaskFreq, &gSfxDefaultFreqAndVolScale, &sGiantsMaskReverbAdd);
+}
+
+void Audio_PlaySfx_GiantsMask(Vec3f* pos, u16 sfxId) {
+    AudioSfx_PlaySfx((sfxId & 0x681F) + 0x20, pos, 4, &sGiantsMaskFreq, &gSfxDefaultFreqAndVolScale,
+                     &sGiantsMaskReverbAdd);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019F638.s")
+void Audio_PlaySfx_Randomized(Vec3f* pos, u16 baseSfxId, u8 randLim) {
+    u8 offset = AudioThread_NextRandom() % randLim;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019F780.s")
+    AudioSfx_PlaySfx(baseSfxId + offset, pos, 4, &gSfxDefaultFreqAndVolScale, &gSfxDefaultFreqAndVolScale,
+                     &gSfxDefaultReverb);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019F7D8.s")
+/**
+ * Plays increasingly high-pitched sword charging sfx as Player charges up the sword
+ */
+void Audio_PlaySfx_SwordCharge(Vec3f* pos, u8 chargeLevel) {
+    chargeLevel &= 3;
```

Comment:
I wonder if it would make a bit more sense with a `% 4` instead (as long as it matches too).

Reply:
%4 makes sense, you just need to put `4U` on the literal, but I think it's still better